tests/unit/test_duckdb_sql.py

```
@@ -0,0 +1,137 @@
+):
+    mock_import_extension.side_effect = Exception("Failed to import extension")
+
+    with fresh_duckdb_connection() as conn:
+        assert conn is not None
+        result = conn.execute(
+            "SELECT extension_name, loaded FROM duckdb_extensions()"
+        ).df()
+        assert result is not None
+        # check that spatial and excel extensions are not loaded as import extension failed
+        result = result[result["extension_name"].isin(["spatial", "excel"])]
+        assert result["loaded"].all() == False
```

Ruff (0.14.5)

Avoid equality comparisons to `False`; use `not result["loaded"].all():` for false checks

Replace with `not result["loaded"].all()`

(E712)